packages/typescript-estree/tests/lib/parse.moduleResolver.default-program-error.test.ts

```diff
@@ -0,0 +1,93 @@
+import { join, resolve } from 'path';
+
+import * as parser from '../../src';
+import type * as astConverterModule from '../../src/ast-converter';
+import type * as sharedParserUtilsModule from '../../src/create-program/shared';
+import type { TSESTreeOptions } from '../../src/parser-options';
+
+const FIXTURES_DIR = join(__dirname, '../fixtures/simpleProject');
+
+// we can't spy on the exports of an ES module - so we instead have to mock the entire module
```

Comment:
😞 these mocks are now repeated ~exactly across five files. That's not very maintainable - if we were to need to change the logic, it'd have to apply to all five files.

I tried commenting out both `jest.mock('../../src/` mocks in this file and this file's tests still passed locally. So at the very least, some of these module mocks can be deleted.

Reply:
Totally agree I need to remove repeated ones!

Let me go through all of them, comment and if tests are passed, I will remove mocks. Otherwise, I will move them to separate utility functions and will reuse them where needed

Reply:
Resolved it in https://github.com/typescript-eslint/typescript-eslint/pull/6092/commits/f4abd7e17222ac0832b744598e2c1f21e8a93d8f, https://github.com/typescript-eslint/typescript-eslint/pull/6092/commits/a09427ff1f798267992cc8ec47ae61b07fd9858c, https://github.com/typescript-eslint/typescript-eslint/pull/6092/commits/f0981ccaff766e039461735a2b35314ca27d994c & https://github.com/typescript-eslint/typescript-eslint/pull/6092/commits/19ee315dd06ae5fbc47c083c0ca2800b24685b0b

Reply:
I only left mocks in main `parse.test.ts`